File: javascript/Hashing/0238. Product of Array Except Self.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var productExceptSelf = function(nums) {
    let res=[]
    let prefix=1
    for(let i=0;i<nums.length;i++){
        res[i]=prefix;
        prefix=prefix*nums[i]
    }
    let post=1;
    for(let i=nums.length-1;i>=0;i--){
        res[i]=res[i]*post;
        post=post*nums[i];
    }
    return res;
};

File: C++/Hashing/0238. Product of Array Except Self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n=nums.size();
        vector<int>res(n);
        int prefix=1;
        for(int i=0;i<n;i++){
            res[i]=prefix;
            prefix=prefix * nums[i];
        }
        int post=1;
        for(int j=n-1;j>=0;j--){
            res[j]=res[j]*post;
            post=post * nums[j];
        }
        return res;
    }
};